### PR TITLE
HV: remove tinycrypt

### PR DESCRIPTION
**Pull request by YadongQi**
Remove tinycrypt files:
Currently, hkdf_sha256 (based on Tinycrypt) is used by Trusty seed
derivation. And we plan to remove Tinycrypt from ACRN hypervisor
now, and replace it with another crypto lib in the future.

Signed-off-by: Qi Yadong <yadong.qi@intel.com>

### CODE REVIEW

**Review by kwang13 — APPROVED** [review]

**Comment by bingzhux** [comment]
update title with this.

Remove tinycrypt files: 
Currently, hkdf_sha256 (based on Tinycrypt) is used by Trusty seed derivation. And we plan to remove Tinycrypt from ACRN hypervisor now, and replace it with another crypto lib in the future.

File: hypervisor/lib/crypto/hkdf.c [hunk 1]
```diff
@@ -30,134 +30,19 @@
...
+	UNUSED(salt_len);
+	UNUSED(info);
+	UNUSED(info_len);
 
-	if (!hkdf_sha256_expand(out_key, out_len,
-				prk, prk_len,
-				info, info_len)) {
-		return 0;
-	}
+	memcpy_s(out_key, out_len, secret, secret_len);
```
Comment (bingzhux) [inline hypervisor/lib/crypto/hkdf.c]: add comments here:
FIXME: currently, we only support one AaaG/Trusty instance, so just simply copy the h/w seed to Trusty. In the future, we will choose another crypto library to derive multiple seeds in order to support multiple AaaG/Trusty instances.

**Review by bingzhux — CHANGES_REQUESTED** [review]

File: hypervisor/lib/crypto/hkdf.c (hunk 1 above)
Comment (YadongQi) [reply hypervisor/lib/crypto/hkdf.c]: done

**Comment by YadongQi** [comment]
PR to upstream: https://github.com/projectacrn/acrn-hypervisor/pull/221
close this one.

**Pull request closed without merging**